test/integration_test.py

```diff
@@ -288,11 +289,20 @@ def test_snapshot_listing(self):
         """
         Test with a small page size to be sure paging works
         """
+        snapshots = self._public_tdr_client.snapshot_names_by_id()
```

Comment:
Not a good idea to base test parameters on the function under test. Think about this some more, and in PL, discuss an your alternative design prior to implementing it.

Reply:
discussed alternatives in PL.